src/compositions/useDrawer.ts

```diff
@@ -0,0 +1,31 @@
+import { computed, Ref } from 'vue'
+import { useBooleanToggle } from '@/compositions/useBooleanToggle'
+
+type UseDrawer = {
+  value: Ref<boolean>,
+  isOpen: Ref<boolean>,
+  open: () => void,
+  close: () => void,
+  toggle: () => void,
+}
+
+/**
+ * The useDrawer composition takes a boolean ref and returns a set of methods
```

Comment:
I'm personally in favor of it for pure TS/JS things like this - while it's not _technically_ needed with TS because the code is self-documenting, this allows us to provide a better dev x when someone isn't as familiar with a composition, how it can be used, and what they return.